paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/partition/PartitionMarkDoneTrigger.java

```diff
@@ -65,21 +78,30 @@ public class PartitionMarkDoneTrigger {
     @Nullable private final Long idleTime;
     private final boolean markDoneWhenEndInput;
     private final Map<String, Long> pendingPartitions;
+    private final List<PartitionMarkDoneAction> actions;
+    private final boolean waitCompaction;
+    private final InternalRowPartitionComputer partitionComputer;
 
     public PartitionMarkDoneTrigger(
             State state,
             PartitionTimeExtractor timeExtractor,
             @Nullable Duration timeInterval,
             @Nullable Duration idleTime,
-            boolean markDoneWhenEndInput)
+            boolean markDoneWhenEndInput,
```

Comment:
Can you keep `PartitionMarkDone` as it is?

Reply:
I reset the commit, and updated in https://github.com/apache/paimon/pull/4398